assets/src/edit-story/app/prepublish/guidance/media.js

```diff
@@ -0,0 +1,140 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { PAGE_HEIGHT, PAGE_WIDTH } from '../../../constants';
+
+const SAFE_ZONE_AREA = PAGE_HEIGHT * PAGE_WIDTH;
+
+const MIN_PIXEL_DENSITY = 2;
+
+const MAX_VIDEO_LENGTH_SECONDS = 60;
+const MAX_VIDEO_WIDTH = 3840;
+const MAX_VIDEO_HEIGHT = 2160;
+const MIN_VIDEO_HEIGHT = 480;
+const MIN_VIDEO_WIDTH = 852;
+
+// todo: get video frames per second
+// MIN_VIDEO_FPS = 24;
+// export function videoElementFps(element) {}
+
+export function mediaElementSizeOnPage(element) {
+  // triggered if only one video is on the page and it takes less than 50% of the safe zone area
+  // encourage users to consider a more immersive media sizing and cropping.
+  const elementArea = element.height * element.width;
+  const isTooSmallOnPage = SAFE_ZONE_AREA / 2 > elementArea;
+
+  if (isTooSmallOnPage) {
+    return {
+      type: 'guidance',
+      elementId: element.id,
+      message:
+        element.type === 'video'
+          ? __(`Video is too small on the page`, 'web-stories')
+          : __(`Image is too small on the page`, 'web-stories'),
+    };
+  }
+
+  return undefined;
+}
+
+export function videoElementSizeOnPage(page) {
+  const videoElementsOnPage = page.elements.filter(
+    ({ type }) => type === 'video'
+  );
+  if (videoElementsOnPage.length === 1) {
+    const [videoElement] = videoElementsOnPage;
+    return mediaElementSizeOnPage(videoElement);
+  }
+  return undefined;
+}
+
+export function mediaElementResolution(element) {
+  if (element.type === 'video') {
+    return videoElementResolution(element);
+  }
+
+  if (element.type === 'image' || element.type === 'gif') {
+    return imageElementResolution(element);
+  }
+
+  return undefined;
+}
+
+function videoElementResolution(element) {
+  const videoResolutionLow =
+    element.resource.full.height <= MIN_VIDEO_HEIGHT &&
+    element.resource.full.width <= MIN_VIDEO_WIDTH;
+  const videoResolutionHigh =
+    element.resource.full.height >= MAX_VIDEO_HEIGHT &&
+    element.resource.full.width >= MAX_VIDEO_WIDTH;
+
+  if (videoResolutionHigh) {
+    return {
+      type: 'guidance',
+      elementId: element.id,
+      message: __(
+        "Video's resolution is too high to display on most mobile devices (>4k)",
+        'web-stories'
+      ),
+    };
+  }
+
+  if (videoResolutionLow) {
+    return {
+      type: 'guidance',
+      elementId: element.id,
+      message: __('Video has low resolution', 'web-stories'),
+    };
+  }
+
+  return undefined;
+}
+
+function imageElementResolution(element) {
```

Comment:
I am positive I am doing this wrong. Here's the actual requirement:
> Low image resolution: actual image asset on screen, at the current zoom, offers <2x pixel density (guideline is to strive for >828 x 1792)

Should I check the dimensions of the image resource against the min requirements here (like so for video resolution)? I'm not sure how to incorporate "current zoom".

Reply:
I would expect "current zoom" to just be the element's width and height (as opposed to the resource width and height). I think that the math here is wrong though since the area of a 2x image is actually 4x (2x horizontal and 2x vertical).

It might be an edge case, but since images could be skewed and stretched, I think that this check probably should check width and height separately to verify both dimensions are at 2x resolution.

Reply:
We still want to incorporate the resource height/width too, right? thinking about it this way:

_if the element height/width is smaller than the resource's height/width:_
is the height or width of the element 2x smaller than the resource's height or width? pixel density is good ✔
otherwise, return this guidance

_if the element height/width is smaller than the resource's height/width:_
it'll never have the required pixel density, so always return this guidance

Reply:
Yes resource height and width are also important. Both of your scenarios are the same I think there might be a typo in your comment. I think the asset just needs to be 2x width and 2x height of the element that it's contained within.